### PR TITLE
Adding blog post for single-spa parcels.

### PR DESCRIPTION
**Pull request by internettrans**

### CODE REVIEW

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: single-spa parcels, explained
+author: Joel Denning
+authorURL: https://twitter.com/joelbdenning
+authorImageURL: https://avatars2.githubusercontent.com/u/5524384?s=400&u=ff145fcb2ae5305555628a446e9f725d4e145aaa&v=4
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:5]: Does this work for you? I haven't been able to figure out how to get the photo to actually render.

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+---
+Ever since single-spa@1.0.0, the single-spa team has been dedicated to bringing microservices to the frontend. We have made it possible for AngularJS, React, Angular, Vue, and other frameworks to coexist side by side in the same page.
+
+And with the release of [version 4](https://github.com/CanopyTax/single-spa/releases/tag/v4.0.0), I’m pleased to announce that [single-spa](https://github.com/CanopyTax/single-spa) is expanding that effort so that ***individual components*** written with different frameworks can interoperate. It is new terrain for the single-spa community, which previously had focused on getting large applications to interoperate with each other, instead of the individual components.
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:9]: Should we link to single-spa.js.org here?

File: website/blog/2018-06-19-single-spa-parcels-explained.md (hunk 1 above)
Comment (internettrans) [reply website/blog/2018-06-19-single-spa-parcels-explained.md:5]: no it doesn't do anything afaict. We'd have to dive into https://docusaurus.io/docs/en/blog to figure it out I think

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Another way to do framework agnostic components?
+
+For those familiar with [web components](https://developer.mozilla.org/en-US/docs/Web/Web_Components) and [custom elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements), you may be wondering why a javascript library would try to do what browsers are starting natively to do.
+
+And as one of the contributors to the custom elements polyfill, let me be the first one to say that we did not make this decision lightly.
+
+If you’re interested in diving into the details, check out [One Company’s Relationship With Custom Elements](https://medium.com/canopy-tax/one-companys-relationship-with-custom-elements-d360baf3b253), which explains some of the difficulties we’ve been through with web components and custom elements.
+
+TLDR: React and some other frameworks don’t interop with custom elements very well. Additionally dealing with inner html, attributes vs properties, and customized builtins can be a pain.
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:19]: I like this. Good idea to include the TLDR.

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Okay but you haven’t told me what a single-spa parcel is
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:21]: Reword: "Okay, but what is a single-spa parcel?"

File: website/blog/2018-06-19-single-spa-parcels-explained.md (hunk 2 above)
Comment (internettrans) [reply website/blog/2018-06-19-single-spa-parcels-explained.md:9]: Well if they're already on the blog that means they're already on single-spa.js.org, so linking to it seems sorta redundant. I am thinking a link to the github page might be nice for people who are looking for the code

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+A parcel is single-spa’s way of building a component in one framework and using it in another.
+
+To implement a parcel, just create a javascript object that has 3–4 functions on it. We call this javascript object a *parcel config* and there are three required functions to implement: bootstrap, mount, and unmount. A fourth function, update, is optional.
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:25]: Possibly link to the docs here so people can read more about what the bootstrap, mount and unmount functions are doing.

File: website/blog/2018-06-19-single-spa-parcels-explained.md (hunk 4 above)
Comment (internettrans) [reply website/blog/2018-06-19-single-spa-parcels-explained.md:21]: The content here is exactly the same as it is on my medium article. I'm not really wanting to have two different versions of the blog post -- it's just a copy paste job

File: website/blog/2018-06-19-single-spa-parcels-explained.md (hunk 5 above)
Comment (internettrans) [reply website/blog/2018-06-19-single-spa-parcels-explained.md:25]: Same as above - this is a copy/paste job from medium and i'm not really wanting to have two different versions of the blog post.

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+Each of the functions will be called by single-spa at the right time, but the parcel config will control what happens. In other words, single-spa controls the “when,” but the parcel config controls the “what” and the “how.”
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:27]: I understand what you're saying here, but someone unfamiliar with single-spa might not. I would recommend rewording the first sentence to give a more detailed description of what you mean by "the right time".

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+Once you’ve implemented the parcel config, simply call singleSpa.mountRootParcel(parcelConfig, parcelProps) to mount it. This is the key to what makes parcels framework agnostic — regardless of whether the parcel config is implemented with React, Angular, Vue, or anything else, to use the parcel you always just call mountRootParcel().
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:29]: I'd suggesting adding a link to the docs for 'mountRootParcel()'

File: website/blog/2018-06-19-single-spa-parcels-explained.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+We’ve glossed over a few things that I want to touch on real quick:
+
+* **How do you implement the lifecycle functions on the parcel config?**
+
+		Use a helper library for your framework of choice. [single-spa-react](https://github.com/CanopyTax/single-spa-react), [single-spa-angular2](https://github.com/CanopyTax/single-spa-angular2) (for angular@2+), [single-spa-angular1](https://github.com/CanopyTax/single-spa-angular1), [single-spa-vue](https://github.com/CanopyTax/single-spa-vue), and [others](https://github.com/CanopyTax/single-spa/blob/master/docs/single-spa-ecosystem.md) will implement the entire parcel config for you.
+
+* **What are the props you pass to mountRootParcel()?**
+
+		The props passed as the second argument to singleSpa.mountRootParcel(parcelConfig, parcelProps) are an object with one required prop and as many custom props as you’d like. The required prop is domElement, which tells the parcel where to mount. And the custom props get passed through to the parcel config lifecycle functions.
```
Comment (alocke12992) [inline website/blog/2018-06-19-single-spa-parcels-explained.md:41]: "is an object" vs "are an object".
I might also recommend rewriting the second portion of that sentence to make it easier to read. 
"... is an object which requires one prop, but can take as many custom props as you'd like to add."

File: website/blog/2018-06-19-single-spa-parcels-explained.md (hunk 1 above)
Comment (frehner) [reply website/blog/2018-06-19-single-spa-parcels-explained.md:5]: You need to update docusaurus to the latest version: it appears you're on `1.0.10` and the latest is `1.2.1`. The authorImageURL attribute was added in `1.1.0`

There have also been 12 updates between the version we're on and the latest version. https://github.com/facebook/Docusaurus/blob/master/CHANGELOG.md#121---2018-06-17

You may also be excited to learn that live reloading is a feature as well.

TLDR; update docusaurus.
